docs/WASI-design-principles.md

```diff
@@ -0,0 +1,95 @@
+# WASI Design Principles
+
```

Comment:
Maybe just call this file `design-principles.md`?   It should be obvious from the repo it lives that its about WASI.  Otherwise every file in this directory will end being called WASI-something, which I think we want to above since its redundant.

Reply:
Done.